js/carrito.js

```diff
@@ -0,0 +1,118 @@
+export class Carrito {
+    constructor(catalogo) {
+        this.catalogo = catalogo;
+        this.productos = [];
+    }
+
+    // Buscar producto en el carrito por el id usando for
+    buscarEnCarrito(id) {
+        for (let i = 0; i < this.productos.length; i++) {
+            if (this.productos[i].producto.id === id) {
+                return this.productos[i];
+            }
+        }
+        return null;
+    }
+
+    // Mostrar catalogo
+    mostrarCatalogo() {
+        let mensaje = "Catalogo de productos \n";
+        this.catalogo.forEach((p) => {
+            mensaje += `ID: ${p.id} | ${p.nombre} | $${p.precio} | Stock: ${p.cantidad}\n`;
+        });
+        alert(mensaje);
+    }
+
+    // Agregar producto al carrito 
+    agregarAlCarrito() {
+        this.mostrarCatalogo();
+        let idIngresado = prompt("Ingrese el ID del producto que desea agregar (o '0' para cancelar):")
+        if (idIngresado === null || idIngresado === "0") return;
+
+        const id = Number(idIngresado);
+        let producto = null;
+
+        for (let i = 0; i < this.catalogo.length; i++) {
+            if (this.catalogo[i].id === id) {
+                producto = this.catalogo[i];
+                break;
+            }
+        }
+
+        if (!producto) {
+            alert("ID invalido.");
+            return;
+        }
+
+        let maxCantidad = producto.cantidad;
+        let cantidadIngresada = prompt(`Cuantas unidades de "${producto.nombre}" desea agregar? (Maximo: ${producto.cantidad})`)
+        if (cantidadIngresada === null) return;
+        const cantidad = Number(cantidadIngresada);
+
+        if (isNaN(cantidad) || cantidad < 1 || cantidad > maxCantidad) {
+            alert("Cantidad Invalida.");
+            return;
+        }
+
+        let item = this.buscarEnCarrito(producto.id);
+        if (item) {
+            if (item.cantidad + cantidad > maxCantidad) {
+                alert("No hay suficiente stock para agregar esa cantidad");
+                return;
+            }
+            item.cantidad += cantidad;
+        } else {
+            this.productos.push({ producto, cantidad });
+        }
+        alert(`Agregaste ${cantidad} x ${producto.nombre} al carrito`);
+    }
+
+    // Mostrar carrito y total
+    mostrarCarrito() {
```

Comment:
Podes pensar en dos métodos de negocio, uno que traiga el carrito y otro que obtenga el total, dejando la interacción de prompts y alerts al main principal. Por ejemplo:
```suggestion
calcularTotal() {
    let total = 0;
    for (let i = 0; i < this.productos.length; i++) {
        const item = this.productos[i];
        total += item.cantidad * item.producto.precio;
    }
    return total;
}
```